idaes/models_extra/power_generation/unit_models/cross_flow_heat_exchanger_1D.py

```diff
@@ -0,0 +1,808 @@
+            )
+            tube_units = (
+                self.config.cold_side.property_package.get_metadata().derived_units
+            )
+        else:
+            shell = self.cold_side
+            tube = self.hot_side
+            shell_units = (
+                self.config.cold_side.property_package.get_metadata().derived_units
+            )
+            tube_units = (
+                self.config.hot_side.property_package.get_metadata().derived_units
+            )
+        if (
+            len(self.config.hot_side.property_package.phase_list) != 1
+            or len(self.config.cold_side.property_package.phase_list) != 1
+        ):
+            raise ConfigurationError(
+                "The CrossFlowHeatExchanger1D model is valid only for property packages "
+                f"with a single phase. Found {len(self.config.hot_side.property_package.phase_list)} "
+                f"phases on the hot side and {len(self.config.cold_side.property_package.phase_list)} "
+                "phases on the cold side."
+            )
+
+        p_hot = self.config.hot_side.property_package.phase_list.at(1)
+        pobj_hot = self.config.hot_side.property_package.get_phase(p_hot)
+        p_cold = self.config.cold_side.property_package.phase_list.at(1)
+        pobj_cold = self.config.cold_side.property_package.get_phase(p_cold)
+        if not pobj_hot.is_vapor_phase():
+            raise ConfigurationError(
+                "The CrossFlowHeatExchanger1D model is valid only for property packages "
+                "whose single phase is a vapor phase. The hot side phase is not a vapor phase."
+            )
+        if not pobj_cold.is_vapor_phase():
+            raise ConfigurationError(
+                "The CrossFlowHeatExchanger1D model is valid only for property packages "
+                "whose single phase is a vapor phase. The cold side phase is not a vapor phase."
+            )
+
+        # Reference
+        add_object_reference(self, "heat_tube", tube.heat)
+        add_object_reference(self, "heat_shell", shell.heat)
+
+        shell_has_pressure_change = False
+        tube_has_pressure_change = False
+
+        if self.config.cold_side.has_pressure_change:
+            if self.config.shell_is_hot:
+                add_object_reference(self, "deltaP_tube", tube.deltaP)
+                tube_has_pressure_change = True
+            else:
+                add_object_reference(self, "deltaP_shell", shell.deltaP)
+                shell_has_pressure_change = True
+        if self.config.hot_side.has_pressure_change:
+            if self.config.shell_is_hot:
+                add_object_reference(self, "deltaP_shell", shell.deltaP)
+                shell_has_pressure_change = True
+            else:
+                add_object_reference(self, "deltaP_tube", tube.deltaP)
+                tube_has_pressure_change = True
+
+        heat_exchanger_common._make_performance_common(  # pylint: disable=W0212
```

Comment:
Minor comment - could you use the full name of the pylint exclusion. It makes it a lot easier to understand what is being skipped and helps for future maintenance.